iroh-relay/src/server/client.rs

```diff
@@ -557,6 +567,41 @@ impl Sink<Frame> for RateLimitedRelayedStream {
     }
 }
 
+/// Counts how many `NodeId`s seen, how many times.
+/// Gets reset every day.
+#[derive(Debug)]
+struct ClientCounter {
+    clients: HashMap<NodeId, usize>,
+    last_clear_date: Date,
+}
+
+impl Default for ClientCounter {
+    fn default() -> Self {
+        Self {
+            clients: HashMap::new(),
+            last_clear_date: OffsetDateTime::now_utc().date(),
+        }
+    }
+}
+
+impl ClientCounter {
+    fn check_and_clear(&mut self) {
+        let today = OffsetDateTime::now_utc().date();
+        if today != self.last_clear_date {
+            self.clients.clear();
+            self.last_clear_date = today;
+        }
+    }
+
+    /// Updates the client counter.
+    fn update(&mut self, client: NodeId) -> bool {
+        self.check_and_clear();
+        let new_conn = !self.clients.contains_key(&client);
+        let _ = self.clients.entry(client).or_insert(0);
+        new_conn
```

Comment:
You can probably do this while only accessing the map once:
```suggestion
        let count = self.clients.entry(client).or_insert(0);
        count == 0
```

I think that's the equivalent of what you had.  But as my other comment says, I'm not sure that's what you want or what you meant.

Reply:
Good point, this is a relic from the old counter. Simplified it to a `HashSet`.